Formula/clojure-lsp.rb

```diff
@@ -0,0 +1,20 @@
+class ClojureLsp < Formula
+  desc "Language Server (LSP) for Clojure"
+  homepage "https://github.com/snoe/clojure-lsp"
+  url "https://github.com/snoe/clojure-lsp/archive/release-20190408T040839.tar.gz"
+  sha256 "79c6d812a8ef4af2cfdd78c4b9aa96674ff9fb8dfeb27869215caa4aee954fae"
+  head "https://github.com/snoe/clojure-lsp.git"
+
+  depends_on "leiningen" => :build
+
+  def install
+    system "lein", "uberjar"
+    jar = Dir["target/clojure-lsp-*-standalone.jar"][0]
+    libexec.install jar
+    bin.write_jar_script libexec/File.basename(jar), "clojure-lsp"
+  end
+
+  test do
+    system "#{bin}/clojure-lsp", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (`foo --version` or `foo --help`) are not sufficient, as explained in the [formula cookbook](https://docs.brew.sh/Formula-Cookbook).

In most cases, a good test would involve running a simple test case: run `#{bin}/foo input.txt`.
- Then you can check that the output is as expected (with `assert_equal` or `assert_match` on the output of `shell_output`)
- You can also check that an output file was created, if that is expected: `assert_predicate testpath/"output.txt", :exist?`

Some advice for specific cases:
- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Yeah I thought bad test was better than no test. :D

This app runs continuously and reads from stdin and writes to stdout. Could you point me to an example formula that tests something similar and eventually kills the process after the test?

Reply:
@markx typically these sorts of tests use `expect` grep/rg/ag on `expect` in the Taps/homebrew/homebrew-core/Formula directory, or checkout percol: `brew cat percol`

Reply:
Updated. Please take another look.

Reply:
LGTM. Going to test locally.